### ISSUE TITLE
trying to build / fail with init-tools.cmd

### ISSUE THREAD

**Issue by smoothdeveloper** [Microsoft/visualfsharp#1140]
I'm getting issues using `build.cmd` and the part related to restoring dotnet core tools.

On shaky connection, the naive download via powershell won't work, and I had to try several times before dotnet-dev-win-x64.1.0.0-beta-002173.zip would eventually arrive completely (downloading same file from browser works well).

After this part is completed, the process gives me pages of messages looking like this, I abort the process because it seems it could go for a long time without giving me any clue about what is going wrong:

```shell-session
C:\dev\src\projects\github.com\visualfsharp>call C:\dev\src\projects\github.com\visualfsharp\init-tools.cmd
The directory is not empty.
Installing dotnet cli...
Restoring BuildTools version 1.0.25-prerelease-00317-01...
Initializing BuildTools ...
Object reference not set to an instance of an object.
C:\dev\src\projects\github.com\visualfsharp\packages\Microsoft.DotNet.BuildTools\1.0.25-prerelease-00317-01\lib\tool-runtime\project.json(44,46): error NU1001: The dependency Microsoft.NETCore.Windows.ApiSets >= 1.0.1-rc2-23923 could not be resolved.
<SNIP>                                                                                                                                                                                                
info : An error occurred while sending the request.                                                                                                                                                     
info :   The operation timed out                                                                                                                                                                        
info : An error was encountered when fetching 'GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/system.linq.parallel/index.json'. The request will now b
e retried.                                                                                                                                                                                              
info : An error occurred while sending the request.                                                                                                                                                     
info :   The operation timed out                                                                                                                                                                        
info : An error was encountered when fetching 'GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/system.io.unmanagedmemorystream/index.json'. The request will 
now be retried.                                                                                                                                                                                         
info : An error occurred while sending the request.                                                                                                                                                     
info :   The operation timed out                                                                                                                                                                        
info : An error was encountered when fetching 'GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/system.linq.expressions/index.json'. The request will no
w be retried.                                                                                                                                                                                           
info : An error occurred while sending the request.                                                                                                                                                     
info :   The operation timed out                                                                                                                                                                        
info : An error was encountered when fetching 'GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/system.linq.queryable/index.json'. The request will now 
be retried.                                                                                                                                                                                             
info : An error occurred while sending the request.                                                                                                                                                     
info :   The operation timed out                                                                                                                                                                        
info :   GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/system.globalization/index.json                                                               
info :   GET https://www.nuget.org/api/v2/FindPackagesById()?id='System.Globalization'                                                                                                                  
info :   GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/system.globalization.extensions/index.json                                                          
info : An error was encountered when fetching 'GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/system.net.http/index.json'. The request will now be ret
ried.                                                                                                                                                                                                   
info : An error occurred while sending the request.                                                                                                                                                     
info :   The operation timed out                                                                                                                                                                        
info :   GET https://www.nuget.org/api/v2/FindPackagesById()?id='System.IO.Compression.ZipFile'                                                                                                         
info :   GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/system.io.compression/index.json                                                                    
info : An error was encountered when fetching 'GET https://www.nuget.org/api/v2/FindPackagesById()?id='System.Linq.Queryable''. The request will now be retried.                                        
info : An error occurred while sending the request.                                                                                                                                                     
info :   The operation timed out                                                                                                                                                                        
info :   GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/system.io.filesystem/index.json                                                                     
info :   GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/system.io.filesystem.primitives/index.json                                                          
info : An error was encountered when fetching 'GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/system.reflection.dispatchproxy/index.json'. The request will 
now be retried.                                                                                                                                                                                         
info : An error occurred while sending the request.                                                                                                                                                     
info :   The operation timed out                                                                                                                                                                        
info : An error was encountered when fetching 'GET https://www.nuget.org/api/v2/FindPackagesById()?id='System.Reflection.Metadata''. The request will now be retried.                                   
info : An error occurred while sending the request.                                                                                                                                                     
info :   The operation timed out                                                                                                                                                                        
info :   GET https://www.nuget.org/api/v2/FindPackagesById()?id='System.Linq'                                                                                                                           
info :   GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/system.linq.parallel/index.json                                                                     
info : An error was encountered when fetching 'GET https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/system.reflection.typeextensions/index.json'. The request will now be retr
ied.                                                                                                                                                                                                    
info : An error occurred while sending the request.                                                                                                                                                     
info :   The operation timed out                                                                                                                                                                        
info :   GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/system.linq.parallel/index.json                                                               
info :   GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/system.linq.expressions/index.json                                                            
info :   GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/system.io.unmanagedmemorystream/index.json                                                          
info :   GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/system.linq.queryable/index.json                                                              
info :   GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/system.net.http/index.json                                                                    
info :   GET https://www.nuget.org/api/v2/FindPackagesById()?id='System.Linq.Queryable'                                                                                                                 
info : An error was encountered when fetching 'GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/system.runtime.interopservices.pinvoke/index.json'. The 
request will now be retried.                                                                                                                                                                            
info : An error occurred while sending the request.                                                                                                                                                     
info :   The operation timed out                                                                                                                                                                        
info : An error was encountered when fetching 'GET https://www.nuget.org/api/v2/FindPackagesById()?id='System.Runtime.Numerics''. The request will now be retried.                                      
info : An error occurred while sending the request.                                                                                                                                                     
info :   The operation timed out                                                                                                                                                                        
info :   GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/system.reflection.dispatchproxy/index.json                                                          
info :   GET https://www.nuget.org/api/v2/FindPackagesById()?id='System.Reflection.Metadata'                                                                                                            
info :   GET https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/system.reflection.typeextensions/index.json                                                                     
info : An error was encountered when fetching 'GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/system.threading.tasks/index.json'. The request will now
 be retried.                                                                                                                                                                                            
info : An error occurred while sending the request.                                                                                                                                                     
info :   The operation timed out                                                                                                                                                                        
info : An error was encountered when fetching 'GET https://www.nuget.org/api/v2/FindPackagesById()?id='System.Threading.Tasks''. The request will now be retried.                                       
info : An error occurred while sending the request.                                                                                                                                                     
info :   The operation timed out                                                                                                                                                                        
info :   GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/system.runtime.interopservices.pinvoke/index.json                                             
info :   GET https://www.nuget.org/api/v2/FindPackagesById()?id='System.Runtime.Numerics'                                                                                                               
info :   OK https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/microsoft.netcore.windows.apisets/1.0.1-rc2-23911/microsoft.netcore.windows.apisets.1.0.1-rc2-23911.n
upkg 7106ms                                                                                                                                                                                             
info :   NotFound https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/system.collections/index.json 762ms                                                      
info : An error was encountered when fetching 'GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/microsoft.fsharp.compiler.netcore/index.json'. The request wil
l now be retried.                                                                                                                                                                                       
info : An error occurred while sending the request.                                                                                                                                                     
info :   The operation timed out                                                                                                                                                                        
info :   GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/system.threading.tasks/index.json                                                             
info :   GET https://www.nuget.org/api/v2/FindPackagesById()?id='System.Threading.Tasks'                                                                                                                
info :   NotFound https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/system.globalization/index.json 702ms                                                    
info :   GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/microsoft.fsharp.compiler.netcore/index.json                                                        
info :   OK https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/system.text.encoding.extensions/index.json 21549ms                                                   
info :   NotFound https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/system.runtime.interopservices.pinvoke/index.json 478ms                                  
info :   GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/system.text.encoding.extensions/4.0.11-rc2-23911/system.text.encoding.extensions.4.0.11-rc2-23911.nu
pkg                                                                                                                                                                                                     
info :   NotFound https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/system.threading.tasks/index.json 507ms                                                  
info :   NotFound https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/system.io/index.json 22463ms                                                                               
info : An error was encountered when fetching 'GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/system.numerics.vectors/index.json'. The request will no
w be retried.                                                                                                                                                                                           
info : An error occurred while sending the request.                                                                                                                                                     
info :   The connection with the server was terminated abnormally                                                                                                                                       
info :   NotFound https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/microsoft.fsharp.compiler.netcore/index.json 646ms                                             
info :   GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/system.numerics.vectors/index.json                                                            
info :   NotFound https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/microsoft.netcore.runtime.coreclr/index.json 23111ms                                                       
info :   OK https://www.nuget.org/api/v2/FindPackagesById()?id='System.IO.FileSystem.Primitives' 22628ms                                                                                                
info :   NotFound https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/system.diagnostics.tracing/index.json 22885ms                                                              
info :   OK https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/system.text.encoding/4.0.11-rc2-23911/system.text.encoding.4.0.11-rc2-23911.nupkg 13628ms            
info :   OK https://www.nuget.org/api/v2/FindPackagesById()?id='System.ComponentModel.Annotations' 23205ms                                                                                              
info :   NotFound https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/system.numerics.vectors/index.json 659ms                                                 
info :   NotFound https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/system.linq.parallel/index.json 1990ms                                                   
info :   NotFound https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/system.net.http/index.json 2024ms                                                        
info :   NotFound https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/system.linq.expressions/index.json 2141ms                                                
info :   OK https://www.nuget.org/api/v2/FindPackagesById()?id='System.IO.UnmanagedMemoryStream' 23628ms                                                                                                
info :   NotFound https://myget-2e16.kxcdn.com/artifacts/fsharp-daily/nuget/v3/flatcontainer/system.globalization.extensions/index.json 23762ms                                                         
info :   OK https://www.nuget.org/api/v2/FindPackagesById()?id='System.Runtime.InteropServices' 23336ms                                                                                                 
info :   OK https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/system.linq/index.json 23732ms                                                                       
info :   GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/system.linq/4.1.0-rc2-23911/system.linq.4.1.0-rc2-23911.nupkg                                       
info :   OK https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/system.reflection.typeextensions/4.1.0-rc2-23911/system.reflection.typeextensions.4.1.0-rc2-23911.nup
kg 12757ms                                                                                                                                                                                              
info :   NotFound https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/system.net.primitives/index.json 24025ms                                                 
info :   OK https://www.nuget.org/api/v2/FindPackagesById()?id='System.Threading.Timer' 23940ms                                                                                                         
info :   OK https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/system.diagnostics.debug/4.0.11-rc2-23911/system.diagnostics.debug.4.0.11-rc2-23911.nupkg 13857ms    
info :   OK https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/system.objectmodel/4.0.12-rc2-23911/system.objectmodel.4.0.12-rc2-23911.nupkg 14519ms                
info :   OK https://www.nuget.org/api/v2/FindPackagesById()?id='System.Reflection' 25417ms                                                                                                              
info :   OK https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/system.reflection.dispatchproxy/index.json 5015ms                                                    
info :   GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/system.reflection.dispatchproxy/4.0.1-rc2-23911/system.reflection.dispatchproxy.4.0.1-rc2-23911.nupk
g                                                                                                                                                                                                       
info :   NotFound https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/system.linq.queryable/index.json 5239ms                                                  
info :   OK https://www.nuget.org/api/v2/FindPackagesById()?id='System.IO.Compression' 27379ms                                                                                                          
info :   OK https://www.nuget.org/api/v2/FindPackagesById()?id='System.Runtime' 27050ms                                                                                                                 
info :   OK https://www.nuget.org/api/v2/FindPackagesById()?id='System.Collections.Concurrent' 27830ms                                                                                                  
</SNIP> Ctrl+C
```

If we are downloading plain nuget packages (even from custom sources) maybe we could define a `paket.dependencies` file and call paket instead, I generally have better experience with it than `dotnet restore` for the moment, it seems `dotnet restore` would just spawn thousands of concurrent downloads and print a lot of stuff.

It would also be very clear what is going to be downloaded, all in a plain text file right in the repository root.

**Comment by smoothdeveloper** [comment]
I ran build.cmd again, after a long while within init-tools.cmd:

```
info :   GET https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/runtime.unix.system.net.nameresolution/4.0.0-rc2-23911/runtime.unix.system.net.nameresolution.4.0.0-rc2-23911.nupkg
info :   OK https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/runtime.unix.system.net.nameresolution/4.0.0-rc2-23911/runtime.unix.system.net.nameresolution.4.0.0-rc2-23911.nupkg 2888ms
log  : Installing runtime.win7-x86.runtime.native.System.IO.Compression 4.0.1-rc2-23911.
log  : Installing runtime.win7-x86.Microsoft.NETCore.Windows.ApiSets 1.0.1-rc2-23911.
log  : Installing runtime.win7-x86.Microsoft.NETCore.Runtime.CoreCLR 1.0.2-rc2-23911.
log  : Installing runtime.win7-x86.Microsoft.NETCore.ConsoleHost 1.0.0-rc2-23911.
log  : Installing runtime.win7.System.Net.NameResolution 4.0.0-rc2-23911.
log  : Installing System.Security.Cryptography.Cng 4.1.0-rc2-23911.
log  : Installing System.Security.Cryptography.Csp 4.0.0-rc2-23911.
log  : Installing runtime.win7.System.Security.Cryptography.Encoding 4.0.0-rc2-23911.
log  : Installing runtime.win7.System.Security.Cryptography.X509Certificates 4.1.0-rc2-23911.
log  : Installing runtime.win7.System.Private.Uri 4.0.1-rc2-23911.
log  : Installing runtime.win7-x64.runtime.native.System.IO.Compression 4.0.1-rc2-23911.
log  : Installing runtime.win7.System.Console 4.0.0-rc2-23911.
log  : Installing runtime.win7-x64.Microsoft.DotNet.CoreHost 0.0.1-beta-00001.
log  : Installing runtime.win7.System.Net.Sockets 4.1.0-rc2-23911.
log  : Installing runtime.win7.Microsoft.Win32.Primitives 4.0.1-rc2-23911.
log  : Installing runtime.win.System.Runtime.InteropServices.RuntimeInformation 4.0.0-rc2-23911.
log  : Installing runtime.osx.10.10-x64.Microsoft.NETCore.ConsoleHost 1.0.0-rc2-23911.
log  : Installing Microsoft.Win32.Registry 4.0.0-rc2-23911.
log  : Installing runtime.osx.10.10-x64.Microsoft.NETCore.Runtime.CoreCLR 1.0.2-rc2-23911.
log  : Installing runtime.any.System.AppContext 4.1.0-rc2-23911.
log  : Installing runtime.any.System.Diagnostics.Tools 4.0.1-rc2-23911.
log  : Installing runtime.any.System.Reflection.Extensions 4.0.1-rc2-23911.
log  : Installing runtime.any.System.Reflection.DispatchProxy 4.0.1-rc2-23911.
log  : Installing System.Security.Principal.Windows 4.0.0-rc2-23911.
log  : Installing runtime.any.System.Runtime.InteropServices 4.1.0-rc2-23911.
log  : Installing runtime.unix.System.Globalization.Extensions 4.0.1-rc2-23911.
log  : Installing runtime.any.System.Reflection.TypeExtensions 4.1.0-rc2-23911.
log  : Installing runtime.any.System.Globalization.Calendars 4.0.1-rc2-23911.
log  : Installing runtime.any.System.Text.Encoding.Extensions 4.0.11-rc2-23911.
log  : Installing runtime.any.System.Threading.Tasks 4.0.11-rc2-23911.
log  : Installing runtime.any.System.Reflection 4.1.0-rc2-23911.
log  : Installing runtime.any.System.Globalization 4.0.11-rc2-23911.
log  : Installing runtime.any.System.IO 4.1.0-rc2-23911.
log  : Installing System.Threading.Overlapped 4.0.1-rc2-23911.
log  : Installing runtime.any.System.Runtime.Handles 4.0.1-rc2-23911.
log  : Installing runtime.any.System.Text.Encoding 4.0.11-rc2-23911.
log  : Installing runtime.any.System.Reflection.Primitives 4.0.1-rc2-23911.
log  : Installing runtime.any.System.Resources.ResourceManager 4.0.1-rc2-23911.
log  : Installing runtime.any.System.Threading.Timer 4.0.1-rc2-23911.
log  : Installing runtime.unix.System.IO.Compression 4.1.0-rc2-23911.
log  : Installing runtime.any.System.Collections 4.0.11-rc2-23911.
log  : Installing runtime.any.System.Linq.Expressions 4.0.11-rc2-23911.
log  : Installing runtime.any.System.Diagnostics.Tracing 4.1.0-rc2-23911.
log  : Installing runtime.unix.System.Runtime.Extensions 4.1.0-rc2-23911.
log  : Installing runtime.osx.10.10.System.Net.NetworkInformation 4.1.0-rc2-23911.
log  : Installing runtime.unix.System.Diagnostics.Debug 4.0.11-rc2-23911.
log  : Installing runtime.osx.10.10.System.Net.Http 4.0.1-rc2-23911.
log  : Installing runtime.unix.System.IO.FileSystem 4.0.1-rc2-23911.
log  : Installing runtime.any.System.Runtime 4.1.0-rc2-23911.
log  : Installing runtime.unix.System.Net.Primitives 4.0.11-rc2-23911.
log  : Installing runtime.osx.10.10.System.Diagnostics.Process 4.1.0-rc2-23911.
log  : Installing runtime.win7.System.Net.Requests 4.0.11-rc2-23911.
log  : Installing runtime.unix.System.Diagnostics.TraceSource 4.0.0-rc2-23911.
log  : Installing runtime.unix.System.Net.Requests 4.0.11-rc2-23911.
log  : Installing runtime.unix.System.Security.Cryptography.Algorithms 4.1.0-rc2-23911.
log  : Installing runtime.native.System.IO.Compression 4.1.0-rc2-23911.
log  : Installing System.Reflection.Emit.Lightweight 4.0.1-rc2-23911.
log  : Installing System.Security.Cryptography.X509Certificates 4.1.0-rc2-23911.
log  : Installing runtime.win7.System.Diagnostics.TraceSource 4.0.0-rc2-23911.
log  : Installing runtime.native.System 4.0.0-rc2-23911.
log  : Installing runtime.native.System.Security.Cryptography 4.0.0-rc2-23911.
log  : Installing System.Diagnostics.DiagnosticSource 4.0.0-rc2-23911.
log  : Installing System.Security.Cryptography.OpenSsl 4.0.0-rc2-23911.
log  : Installing runtime.native.System.Net.Http 4.0.1-rc2-23911.
log  : Installing System.Security.Cryptography.Encoding 4.0.0-rc2-23911.
log  : Installing runtime.unix.System.Console 4.0.0-rc2-23911.
log  : Installing System.Private.Uri 4.0.1-rc2-23911.
log  : Installing runtime.unix.System.Runtime.InteropServices.RuntimeInformation 4.0.0-rc2-23911.
log  : Installing runtime.osx.10.10-x64.Microsoft.DotNet.CoreHost 0.0.1-beta-00001.
log  : Installing runtime.unix.Microsoft.Win32.Primitives 4.0.1-rc2-23911.
log  : Installing runtime.unix.System.Net.Sockets 4.1.0-rc2-23911.
log  : Installing runtime.osx.10.10-x64.runtime.native.System.IO.Compression 1.0.1-rc2-23911.
log  : Installing runtime.win7.System.Diagnostics.Process 4.1.0-rc2-23911.
log  : Installing runtime.osx.10.10-x64.runtime.native.System 1.0.1-rc2-23911.
log  : Installing runtime.osx.10.10-x64.runtime.native.System.Security.Cryptography 1.0.1-rc2-23911.
log  : Installing runtime.unix.System.Security.Cryptography.X509Certificates 4.1.0-rc2-23911.
log  : Installing runtime.osx.10.10-x64.runtime.native.System.Net.Http 1.0.1-rc2-23911.
log  : Installing runtime.unix.System.Security.Cryptography.Encoding 4.0.0-rc2-23911.
log  : Installing runtime.unix.System.Private.Uri 4.0.1-rc2-23911.
log  : Installing System.IO.FileSystem.Watcher 4.0.0-rc2-23911.
log  : Installing System.Net.NameResolution 4.0.0-rc2-23911.
log  : Installing runtime.unix.System.Net.NameResolution 4.0.0-rc2-23911.
log  : Installing runtime.osx.10.10.System.IO.FileSystem.Watcher 4.0.0-rc2-23911.
log  : Installing runtime.ubuntu.14.04-x64.Microsoft.NETCore.ConsoleHost 1.0.0-rc2-23911.
log  : Installing runtime.ubuntu.14.04-x64.Microsoft.NETCore.Runtime.CoreCLR 1.0.2-rc2-23911.
log  : Installing runtime.linux.System.Net.NetworkInformation 4.1.0-rc2-23911.
log  : Installing runtime.linux.System.Net.Http 4.0.1-rc2-23911.
log  : Installing runtime.linux.System.Diagnostics.Process 4.1.0-rc2-23911.
log  : Installing runtime.ubuntu.14.04-x64.Microsoft.DotNet.CoreHost 0.0.1-beta-00001.
log  : Installing runtime.ubuntu.14.04-x64.runtime.native.System.IO.Compression 1.0.1-rc2-23911.
log  : Installing runtime.ubuntu.14.04-x64.runtime.native.System 1.0.1-rc2-23911.
log  : Installing runtime.ubuntu.14.04-x64.runtime.native.System.Net.Http 1.0.1-rc2-23911.
log  : Installing runtime.ubuntu.14.04-x64.runtime.native.System.Security.Cryptography 1.0.1-rc2-23911.
log  : Installing runtime.linux.System.IO.FileSystem.Watcher 4.0.0-rc2-23911.
log  : Installing runtime.win7-x64.Microsoft.NETCore.Runtime.CoreCLR 1.0.2-rc2-23911.
log  : Installing runtime.win7-x64.Microsoft.NETCore.Windows.ApiSets 1.0.1-rc2-23911.
log  : Installing runtime.win7-x64.Microsoft.NETCore.ConsoleHost 1.0.0-rc2-23911.
log  : Installing runtime.win7.System.Globalization.Extensions 4.0.1-rc2-23911.
log  : Installing runtime.win7.System.Runtime.Extensions 4.1.0-rc2-23911.
log  : Installing runtime.win7.System.Net.Primitives 4.0.11-rc2-23911.
log  : Installing runtime.win7.System.IO.Compression 4.1.0-rc2-23911.
log  : Installing runtime.win7.System.Diagnostics.Debug 4.0.11-rc2-23911.
log  : Installing runtime.win7.System.IO.FileSystem 4.0.1-rc2-23911.
log  : Installing runtime.win7.System.Net.NetworkInformation 4.1.0-rc2-23911.
log  : Installing runtime.win7.System.Net.Http 4.0.1-rc2-23911.
log  : Installing runtime.win7.System.Security.Cryptography.Algorithms 4.1.0-rc2-23911.
error: Could not find a part of the path 'C:\Users\gauthier\AppData\Local\Temp\NuGet\TempCache\b1234acb-b44b-4559-884a-dcdb42964009\440uhypk.kmq'.
Error: dotnet restore failed
```

**Comment by dsyme** [comment]
@enricosada @KevinRansom  See also this comment from @smoothdeveloper  asking for us to maintain a .NET 4.x development path for the repo (one which doesn't download or use .NET Core):  https://github.com/Microsoft/visualfsharp/commit/1dfe7029d2e06065ce68e437f393acb6c23a11b2#commitcomment-17318710

**Comment by smoothdeveloper** [comment]
If we don't set `BUILD_NETCORE = 1` we should really not have to go through those steps, it would be great if `build.cmd` could handle this properly.

Also, if the messages I'm seeing are "expected", it might be worth to nudge dotnet/cli team to really trim this down to only informative message by default.

When the msbuild stuff in VS will be fixed (#1144) I'm going to push some more work on prototype FAKE script for simple build, as a separate effort (not to be included in F# repository, but just as a tool for compiler hackers who find the setup / debug daunting); I'm following structure of build.cmd and it really doesn't seem to be that huge of a task (for the basic use cases), I think it has many advantages if we could end up using FAKE build scripts.

**Comment by dsyme** [comment]
> If we don't set BUILD_NETCORE = 1 we should really not have to go through those steps

@smoothdeveloper This has come up.  It's tricky:

* On the one hand we want this repo to soon compile on Linux and Mac as a cross-platform repo by default.  That means using .NET Core.  In that case, it would be strange to have Windows development still using .NET Framework by default

* On the other hand, there's a huge amount of value in still allowing a purely .NET Framework development option on Windows (and potentially a Mono development option on Linux/Mac), for all the reasons you indicate.  

I find the Mac/Linux argument very convincing and I think it's appropriate to make .NET Core the default for this reason.  However It would seem best to also  allow "netfx" as an option to build.cmd, which would avoid any use of .NET Core and to add that to our CI matrix.

Equally I'm not totally sure if we're all up for maintaining the bi-modal existence of a repo like this.   To be honest I think a very large number of .NET repositories are going to face this dilemma.

**Comment by enricosada** [comment]
I think we can fix your issue, so

I see .net core cli ( sdk ) as a build dependency like perl, visual studio sdk etc 

Nobody question perl **as dependency** because is a **one time install**

Dotnetcli instead is downloaded in a subdirectory if not exists, but a `git clean` remove it, so is often downloaded

We already use a specific version of .net cli, so se can

1. Developers **can** install dotnetcli machinewide once, and add to path. Like perl
1. Build.cmd check if the specific version is already installed ( just check `dotnet --version`). If installed we dont need to do nothing. 
1. If build.cmd doesn't find dotnet cli, it can:
    1. Or Stop and ask Dev to install it machinewide
    1. Or install in subdir ( like in ci )


All the nuget packages downloaded by `dotnet restore` are **already cached** in nuget package cache, so it's only the first time when version change. 

We shouldn't remove the netcli , it's going to help a lot xplat. We should learn how to use it. 

Fake is another issue, ortogonal to netcore. 
Script is complex because there a lot of steps. We should simplify.
Fake can only help xplat, we are not going to use it fully, we just run msbuild, no docs/deploy/tag/etc. but xplat add dep to mono. We can just use .netcore fsi and a .fsx file, we already have netcli as dep

**Comment by smoothdeveloper** [comment]
@enricosada, thanks for sharing your thoughts.

Regarding Perl, I don't question it but I don't have it, it is only for `RunTests.cmd` and I could build the compiler without it.

I agree `dotnet.exe` and tools are normal to be a dependency, although if in short future we can move it to where it is necessary only (when building for dotnet core), because the output of it is really misleading and frankly useless (compare that to `paket install` which is doing similar job), I might not be the only one to CTRL+C when I see pages and pages of error messages.

Regarding mono, today it is a more mature runtime, with greater library support, so I think until situation becomes similar with `dotnet.exe` (it's going to take some time), it is fine to rely on it, at least for my effort for scripting (only intended for my personal use and will not land in this repo as far as I can see).

FAKE/.fsx does not only help cross platform IMHO, I have very low skills maintaining and troubleshooting .bat/.sh while it is very easy for any F# developer to use / step through a .fsx script (nobody interested in this repository would do anything with it without such tools installed), it helps overall maintainability by greater number because it is a more flexible solution than plain shell script.

* can we make output of `dotnet restore` any sensible; whatever the way, I don't think I have any chance to get feedback upstream, I didn't get answers for simple questions I made regarding scripting and project.lock.json, the overall feeling is that they are hacking hard to get the platform running, but end-user usability (sane output) and interacting well with OS eco-system / community (helping people to port their stuff) is not their priority
* for windows (not xplat) can we move it to a stage where it is going to be actually used in the script rather than a the very beginning?

**Comment by smoothdeveloper** [comment]
@dsyme thanks too for feedback

> Equally I'm not totally sure if we're all up for maintaining the bi-modal existence of a repo like this. To be honest I think a very large number of .NET repositories are going to face this dilemma.

Maybe some new projects will only support dotnet core, which is fine, but until the new runtime can support all major open source projects and widely used tools (like msbuild to start with), I can foresee that mono will be relied on primarily for xplat support (I believe msbuild is still better tool than what is being done with `project.json`, and it is not going to change that quick).

For visualfsharp repository (and fsharp/fsharp), it is a long way before we can rely on dotnet core primarily/only, and I don't see a hurried push toward this goal as high priority.

I still understand the value of all efforts to bring everything under this repository to the new runtime, and I'm happy that even today people making projects for this platform can rely on `fsc` being easily working.

**Comment by enricosada** [comment]
.net core can help now the build, not net40 only

We are replacing fssrgen/fslexyac etc with netcore version 

I don't think the output of dotnet restore matter, can be less verbose and we can improve it (there are arguments for verbosity => pr)

I dont see people questioning mono for build in osx/unix
This repo is going xplat soon. With netcore. We shoud not waste time fight that, We should facilitate it. 
We should improve developer life ( i use this repo too :smile: ), little step but there is a clear directorion ihmo
